### PR TITLE
OpenJ9 JIT changes to support OMR FrontEnd base class refactoring

### PR DESCRIPTION
**Pull request by 0xdaryl**
Breaking dependence on eclipse/omr#7540 due to file changes.

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins test sanity.functional,sanity.openjdk xlinux,plinux,zlinux,alinux jdk21 depends eclipse/omr#7540

This was tested internally across platform, but just doing a quick sniff test here in the PR prior to review.

**Comment by 0xdaryl** [comment]
Z OpenJDK failure appears to be #17955.

**Comment by 0xdaryl** [comment]
@mstoodle : I'd appreciate a review from you on this

@dsouzai : I'd like a review from you too, and ask that you drive the eventual coordinated merge when it is ready.

**Review by dsouzai — APPROVED** [review]
Changes look fine given the OMR changes it depends on.

**Comment by 0xdaryl** [comment]
Jenkins test sanity.functional xlinux,plinux,zlinux,alinux jdk21 depends eclipse/omr#7540

**Comment by 0xdaryl** [comment]
@dsouzai : The subset of sanity acceptance testing passed, as well as an internal build on SDK8 that included 32-bit testing.  Could you merge the two PRs if you are satisfied please?

File: runtime/compiler/x/env/J9CPU.cpp [hunk 1]
```diff
@@ -25,18 +25,13 @@
...
 #include "codegen/CodeGenerator.hpp"
 #if defined(J9VM_OPT_JITSERVER)
 #include "control/CompilationRuntime.hpp"
 #include "control/CompilationThread.hpp"
 #include "runtime/JITClientSession.hpp"
 #endif /* defined(J9VM_OPT_JITSERVER) */
 
-// This is a workaround to avoid J9_PROJECT_SPECIFIC macros in x/env/OMRCPU.cpp
-// Without this definition, we get an undefined symbol of JITConfig::instance() at runtime
-TR::JitConfig * TR::JitConfig::instance() { return NULL; }
```
Comment (mstoodle) [inline runtime/compiler/x/env/J9CPU.cpp:38]: This one always annoyed me. Glad to see we're finally getting rid of it :) .

**Review by mstoodle — APPROVED** [review]
lgtm! Thanks @0xdaryl !

**Pull request merged**